packages/Icon/src/createIconComponent.js

```diff
@@ -0,0 +1,59 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+export default glyphs => {
+  return class Icon extends PureComponent {
+    static displayName = 'Icon';
+    static propTypes = {
+      glyph: PropTypes.oneOf(Object.keys(glyphs)).isRequired,
+      size: PropTypes.oneOfType([
+        PropTypes.oneOf(['default', 'large', 'xlarge']),
+        PropTypes.number,
+      ]),
+    };
+
+    static defaultProps = {
+      size: 'default',
+    };
+
+    getHumanReadableTitle() {
```

Comment:
So when I called these 'pure functions' I meant that they don't rely on the state of the component instance. These can be written as functions that take an input and give an output - by which I mean they don't need to know anything about the instance state, and as such shouldn't be instance methods.

Having these as instance methods implies to the reader that there is a reliance on internal state, which isn't true.

My desire is these be standalone functions or static methods.

or...

Just inline them in the render function.